src/main/java/com/example/bookstore/mapper/ShoppingCartMapper.java

```diff
@@ -0,0 +1,28 @@
+package com.example.bookstore.mapper;
+
+import com.example.bookstore.dto.cart.items.CartItemDto;
+import com.example.bookstore.dto.shopping.cart.ShoppingCartDto;
+import com.example.bookstore.model.CartItem;
+import com.example.bookstore.model.ShoppingCart;
+import org.mapstruct.InjectionStrategy;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.NullValueCheckStrategy;
+
+@Mapper(componentModel = "spring",
+        injectionStrategy = InjectionStrategy.CONSTRUCTOR,
+        nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS,
+        implementationPackage = "<PACKAGE_NAME>.impl")
+public interface ShoppingCartMapper {
+    @Mapping(source = "user.id", target = "userId")
+    ShoppingCartDto toDto(ShoppingCart shoppingCart);
+
+    default CartItemDto toCartItemDto(CartItem cartItem) {
+        return new CartItemDto(
+                cartItem.getId(),
+                cartItem.getBook().getId(),
+                cartItem.getBook().getTitle(),
+                cartItem.getQuantity()
+        );
+    }
```

Comment:
why default?

you could've used `@Mapping` annotations

Reply:
would it be better to use Mappers.getMapper in default method?

Reply:
@teract10s nope, let's avoid complication, a method with `@Mapping` annotations is fine

Reply:
@NazarSmith228 what do you mean? How can i use Mapping to map collections?